client/js/content.js

```diff
@@ -0,0 +1,102 @@
+const RATINGS = 'ratings';
+
+function getTitleFromCard(elem) {
+    const title = elem.firstElementChild.innerText;
+    return title || '';
+}
+
+const getRatings = async (title) => {
+    const ratingsInLocalStorage = getRatingsFromLocalStorage(title);
+
+    if (ratingsInLocalStorage) {
+        return ratingsInLocalStorage;
+    }
+
+    const titleRatings = await fetchRatings(title);
+    storeRatingsInLocalStorage(title, titleRatings);
+    return titleRatings;
+};
+
+function storeRatingsInLocalStorage(title, ratings) {
+    if (title && ratings) {
+        let ratingsInStorage = JSON.parse(localStorage.getItem(RATINGS));
+        if (!ratingsInStorage) {
+            ratingsInStorage = {};
+        }
+        ratingsInStorage[title] = ratings;
+        localStorage.setItem(RATINGS, JSON.stringify(ratingsInStorage));
+    }
+}
+
+function getRatingsFromLocalStorage(title) {
+    const ratingsInLocalStorage = JSON.parse(localStorage.getItem(RATINGS));
+    const titleRatings = ratingsInLocalStorage && ratingsInLocalStorage[title];
+    return titleRatings;
+}
+
+const getRatingDivElement = (title, rating) => {
+    const div = document.createElement('div');
+    div.innerHTML = `${title} rating: ${rating}`;
+    return div;
+};
+
+const getInnerHTML = (imdbRating, rottenTomatoRating) => {
+    const innerHTML = `
+    <div>
+    <p style="background-color: red;
+        z-index: 2;
+        margin-top: 0px;
+        position: absolute;
+        top:0; right:0;"
+    >IMDB: ${imdbRating}</p>
+    <p style="background-color: red;
+    z-index: auto;
+    margin-top: 0px;
+    position: absolute;
+    top:0; left:0;"
+>Rotten Tomato: ${rottenTomatoRating}</p>
+    </div>
+    `;
+
+    return innerHTML;
+};
+
+const addRatingsToTile = async (movieTileElement) => {
+    const videoTitle = getTitleFromCard(movieTileElement);
+    const tileParent = document.querySelector(
+        '.previewModal--metadatAndControls-container'
+    );
+    const ratings = await getRatings(videoTitle);
+
+    if (!movieTileElement.hasAttribute('ratings')) {
+        movieTileElement.insertAdjacentHTML(
+            'afterend',
+            getInnerHTML(ratings.imdbRating, ratings.rottenTomato)
+        );
+        movieTileElement.setAttribute('ratings', ratings);
+    }
+
+    if (tileParent) {
+        tileParent.appendChild(getRatingDivElement('IMDB', ratings.imdbRating));
+        tileParent.appendChild(
+            getRatingDivElement('Rotten Tomato', ratings.rottenTomato)
+        );
+    }
+};
+
+const induceDelay = (elem, callback) => {
+    elem.onmouseover = () => {
+        setTimeout(() => callback(elem), 2000);
+    };
+};
+
+const observeChangesInDom = async () => {
```

Comment:
We can have an event triggered change detection, say scroll or mouse event (on top of my mind). This works as of now.

Reply:
For the use case we have, we need ratings on each tile when we hover. This takes care of that.

I have another method of mutation tracking which I am exploring as well.

What Eshita has to do, in that we will need these change events. But for the portion I had, this serves the purpose

Reply:
this function calls induceDelay and then within delay mouse events are linked. Please refer induceDelay function